x-pack/legacy/plugins/integrations_manager/public/hooks/links.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { generatePath } from 'react-router-dom';
+import { PLUGIN } from '../../common/constants';
+import { patterns } from '../routes';
+import { useCore } from './index';
+
+// TODO: get this from server/integrations/handlers.ts (move elsewhere?)
+// seems like part of the name@version change
+interface DetailParams {
+  name: string;
+  version: string;
+}
+
+function addBasePath(path: string) {
+  const { http } = useCore();
+  return http.basePath.prepend(path);
+}
+
+function appRoot(path: string) {
+  // include '#' because we're using HashRouter
+  return addBasePath(patterns.APP_ROOT + '#' + path);
+}
+
+export function useLinks() {
+  return {
+    toAssets: (path: string) => addBasePath(`/plugins/${PLUGIN.ID}/assets/${path}`),
+    toListView: () => appRoot(patterns.LIST_VIEW),
+    toDetailView: ({ name, version }: DetailParams) =>
+      appRoot(generatePath(patterns.DETAIL_VIEW, { pkgkey: `${name}-${version}` })),
+  };
```

Comment:
These are all "local" links, right? If we're linking within our own app, we should only need the part of the path after the `#`, since that's all our router cares about, I think?

Reply:
I seemed to have lost my comment about this at some point.

They currently return `/BASE_PATH/app/integrations_manager#/PATH/TO/VIEW`

We could have them return `#/PATH/TO/VIEW` but that assumes their current location/url which makes them less flexible/useful.

This way, the correct/unambiguous path is always returned. It'll work within our app, regardless of how we choose to rearrange things, as well as if other tools want to generate links to the views or embed them.